### ISSUE TITLE
`arbitrary_source_item_ordering`

### ISSUE THREAD

**Issue by amab8901** [rust-lang/rust-clippy#15129]
### Summary

if two items have different integer suffixes but identical everywhere to the left of the integer suffix, then I think the `arbitrary_source_item_ordering` lint should apply the numerical ordering of the entire suffixed integer as a whole number (rather than its individual digits), while continuing to apply alphabetical ordering everywhere to the left of the integer suffix. 

### Lint Name

arbitrary_source_item_ordering

### Reproducer

I tried this code:

```rust
pub(crate) enum ByteType {
    Uint8,
    Uint16,
    Uint32,
    Uint64,
}
```

I saw this happen:

lint triggered:
```
incorrect ordering of items (must be alphabetically ordered)
```

I expected to see this happen:
lint should be silent

### Version

```text
rustc 1.87.0 (17067e9ac 2025-05-09)
binary: rustc
commit-hash: 17067e9ac6d7ecb70e50f92c1944e545188d2359
commit-date: 2025-05-09
host: x86_64-unknown-linux-gnu
release: 1.87.0
LLVM version: 20.1.1
```

**Comment by samueltardieu** [comment]
The current behavior is consistent with the current lint definition ("ordering source code alphabetically"). This is an enhancement proposal, that could require a new configuration option.

@rustbot label -C-bug -I-false-positive +C-enhancement
